js/interactive-guides/step-content.js

```diff
@@ -121,100 +206,108 @@ var stepContent = (function() {
       - if it has, show the existing content
       - otherwise create the new content
       Inputs: {JSON} step
-              {Boolean} navButtonClick: True if they clicked on prev/next buttons and false otherwise
   */
-  var createContents = function(step, navButtonClick) {
-    contentManager.setInstructions(step.name, step.instruction);
-
-    tableofcontents.selectStep(step, navButtonClick);
-    contentManager.setInstructions(step.name, step.instruction);
-
-    __updateTitle(step.title);
-    __updateDescription(step.description);
-    __updateInstructions(step.name);
-
-    __hideContents();
+  var createContents = function(step) {
+    tableofcontents.selectStep(step);
     currentStepName = step.name;
+    __hideContents(step.name); // Hide other steps that are not for this step
 
-    // Highlight the next button if all of the instructions are complete or there are no instructions
-    contentManager.enableNextWhenAllInstructionsComplete(step.name);
-
-    if (!__lookForExistingContents(step)) {
-      if (step.content) {
-        var content = step.content;        
-        var displayTypeCounts = {}; // Map of displayType to the displayCount for that type
-        var defaultBootstrapColSize = "col-sm-12";
-        // two contents will be side by side. Otherwise, it will be stack on top of each other.
-        if (step.content.length == 2) {
-          defaultBootstrapColSize = "col-sm-6";
+    // Check if there is a parent step to load
+    if(step.parent){
+      createContents(step.parent);
+    }
+    else{
+      if (!__lookForExistingContents(step)) {
+        __buildContent(step);
+        if(step.sections){
+          for(var i = 0; i < step.sections.length; i++){
+            contentManager.setInstructions(step.name, step.instruction);
```

Comment:
__buildContent will call contentManager.setInstructions. I think you can remove this call here.

Reply:
One of them is the instruction block (green block) and the other is looking for instruction contents (the different instructions inside of the green block).

Reply:
Done